hashicorp/web-unified-docs
# Feat: add version to nav data

# Changes

- Script to add version to nav data, based off of version metadata. For example, if `v1.9.x` is marked as `isLatest: true` in the version metadata, it will not add versions to that file.


- [Asana task](https://app.asana.com/0/1207899860738460/1208471953509366/f)

# Testing
1. Make a request to this branch's endpoint
```
curl https://web-unified-docs-git-heat-choreadd-version-to-8e2440-hashicorp.vercel.app/api/content/terraform/nav-data/v1.7.x/intro
```
It should look like this:

```
{"result":{"navData":[{"heading":"Introduction to Terraform"},{"title":"What is Terraform?","path":"v1.7.x/"},{"title":"Use Cases","path":"v1.7.x/use-cases"},{"title":"Get Started","href":"https://learn.hashicorp.com/collections/terraform/aws-get-started?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS"},{"title":"Terraform Editions","path":"v1.7.x/terraform-editions"},{"title":"The Core Terraform Workflow","path":"v1.7.x/core-workflow"},{"title":"Terraform vs. Alternatives","routes":[{"title":"Overview","path":"v1.7.x/vs"},{"title":"Chef, Puppet, etc.","path":"v1.7.x/vs/chef-puppet"},{"title":"CloudFormation, Heat, etc.","path":"v1.7.x/vs/cloudformation"},{"title":"Boto, Fog, etc.","path":"v1.7.x/vs/boto"},{"title":"Custom Solutions","path":"v1.7.x/vs/custom"}]}]}}%
```

2. Make a request to the **current Content API** endpoint
```
curl https://content.hashicorp.com/api/content/terraform/nav-data/v1.7.x/intro
```
It should look like this: 
<img width="1258" alt="Screenshot 2024-11-01 at 17 10 31" src="https://github.com/user-attachments/assets/df499cf3-bfae-4150-b7e3-5b28fe10a2ce">

3. The responses should be the same for each endpoint.


4. See that the current UDR API endpoint returns a different response. Make a request with
```
curl https://web-presence-experimental-docs.vercel.app/api/content/terraform/nav-data/v1.7.x/intro
```
It should look like this (notice there are no versions in the paths/hrefs).
<img width="1251" alt="Screenshot 2024-11-01 at 17 12 50" src="https://github.com/user-attachments/assets/96b68916-7c5f-4a9e-885f-a48a65f2777b">
